apps/site/src/_pages/docs/4_spec/02_graph-module.mdx

```diff
@@ -1,303 +1,445 @@
 
-- `entities`
-- `entityTypes`
-- `links`
-- `linkedAggregations`
+## Entity definition
 
-If defined, each field MUST be an array of JSON objects, which MUST conform to the structure defined for each of these resources, as specified in [entity definition](#structure-specification).
+### <Hidden>Entity </Hidden>Structure
 
-##### Embedding applications
+#### <Hidden>Entity Structure </Hidden>Summary
 
-Embedding applications MUST use the constraints set in the block’s schema, if defined, to constrain the data it sends under `properties` in the **block entity**, assuming the embedding application implements the `blockEntity` message.
+An entity is a combination of **metadata** and **properties**.
 
-Embedding applications MAY choose to display their own UI to allow users to set any values on the block entity described in `configProperties`, and could allow users to set values for them for all blocks of a type (rather than on a per-block instance basis)
+The **metadata** is a minimum collection of fields that are fixed and specified in this document, although embedding applications MAY choose to add their own.
 
-## Entity definition
+The **properties** is a JSON object which satisfies the constraints of the entity’s **entity type**.
 
-### Structure
+#### <Hidden>Entity Structure </Hidden>Specification
 
-#### <Hidden>Structure </Hidden>Summary
+##### Entities
 
-An entity is a combination of **identifying fields** and **properties**.
+When embedding applications pass entities to blocks, whether the **block entity** or any other:
 
-The **identifying fields** are fixed and specified in this document, and appear as keys at the root of the entity data object.
+- entities MUST be represented by an object, which:
 
-The **properties** can be any JSON object which is valid for the entity’s type. This object is nested under a `properties` key on the entity data object.
+  - MUST contain a field `metadata`, an object which:
+  - MUST contain a field `recordId`, an object which:
+    - MUST contain a field `entityId` identifying the entity, which MUST be a string.
+    - MUST contain a field `editionId` which identifies the specific _edition_ of the entity, which MUST be a string, and MUST be unique for each edition of an entity for datastores that have versioning for their data.
+  - MUST contain an `entityTypeId` identifying the entity type the entity belongs to, which MUST be a string.
 
-Each entity has an **entity type** representing the type of thing that it is. In turn, an entity type has a **schema** which sets constraints on the permitted **properties** belonging to an entity.
+  - MUST contain a field `properties` if the block has defined any `required` properties in its schema, and otherwise MAY be omitted. If present, it:
```

Comment:
I think our types require this field? we might as well anyway
```suggestion
  - MUST contain a field `properties` which:
```

Reply:
` & (Properties extends null ? {} : { properties: Properties })`

They do not, I'd opt to not change this for now to keep the implementation in sync, we can fix in follow up